types/tx.go

```diff
@@ -229,3 +287,111 @@ func WrapMalleatedTx(originalHash []byte, shareIndex uint32, malleated Tx) (Tx,
 	}
 	return proto.Marshal(&wTx)
 }
+
+// Validate verifies the proof. It returns nil if the proof is valid.
+// Otherwise, it returns a sensible error.
+func (sp SharesProof) Validate() error {
```

Comment:
[blocking docs]
lets leave a comment here or somewhere indidcating that this is tested in the application